release.sh

```diff
@@ -83,9 +83,11 @@ pushd "$TEMP3/benchexec"
 pip install "pip >= 10.0" "setuptools >= 42.0.0, < 58" "wheel >= 0.32.0"
 # avoid the wheel on PyPi for nose, it does not work on Python 3.10
 pip install nose --no-binary :all:
-pip install -e "."
-python setup.py nosetests
-python setup.py sdist bdist_wheel
+# install build if it is not installed by default (it usually is)
+pip install build
+pip install -e ".[dev]"
+python -m nose
+python -m build
```

Comment:
There is also an explanation of what `release.sh` does in `doc/DEVELOPMENT.md`, could you please update it with the reference to `build` and explain that it builds source dist and wheel for us?